### PR TITLE
Use Dto for Run REST endpoint, and add new Project REST API calls

### PR DESCRIPTION
**Pull request by froggleston**
Tested and all seems to be working.

### CODE REVIEW

**Comment by sqwci** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [RunRestController.java#L57](https://github.com/TGAC/miso-lims/blob/be646a5f9cc20572dbd4b9e4900d66852fd8eb7d/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/RunRestController.java#L57): Remove this unused import 'com.fasterxml.jackson.databind.ObjectMapper'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/ProjectRestController.java [hunk 1]
```diff
@@ -69,18 +63,36 @@
...
+
   public void setLibraryService(LibraryService libraryService) {
     this.libraryService = libraryService;
   }
 
+  public void setPoolService(PoolService poolService) {
+    this.poolService = poolService;
+  }
+
+  public void setRunService(RunService RunService) {
```
Comment (sqwci) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/ProjectRestController.java]: MAJOR Remove this unused method parameter "RunService". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
MINOR Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00117)

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/ProjectRestController.java [hunk 2]
```diff
@@ -69,18 +63,36 @@
...
+    this.runService = runService;
```
Comment (sqwci) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/ProjectRestController.java]: MAJOR Remove this useless assignment; "runService" already holds the assigned value along all execution paths. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4165)

**Comment by sqwci** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/ProjectRestController.java (hunk 1 above)
Comment (callunity) [reply miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/ProjectRestController.java]: Typo in param name

**Comment by apmasell** [comment]
(once you fix the thing Heather pointed out)

**Comment by froggleston** [comment]
Fixed typo.

**Pull request merged**